var/spack/repos/builtin/packages/lua/package.py

```diff
@@ -37,6 +39,13 @@ class Lua(Package):
     # luarocks needs unzip for some packages (e.g. lua-luaposix)
     depends_on('unzip', type='run')
 
+    # By default, we are missing lua.pc
+    patch(
+        "http://lua.2524044.n2.nabble.com/attachment/7666421/0/pkg-config.patch",
+        sha256="208316c2564bdd5343fa522f3b230d84bd164058957059838df7df56876cb4ae"
```

Comment:
Sorry, that's what I get for typing in Github instead of trying the code:
```suggestion
        sha256="208316c2564bdd5343fa522f3b230d84bd164058957059838df7df56876cb4ae",
```

Reply:
It's okay I saw this and fixed it! It was this change that actually broke the build - I don't know enough about patching to say why. Maybe it was some kind of permissions issue?

Reply:
Not sure, I tried the same and it works for me:
```console
$ spack -d patch lua+pcfile
[ ... ]
==> [2021-03-22-10:22:20.439018] Installing /tmp/culpo/spack-stage/resource-luarocks-mru76cqdlrwasvbhutudefmwr7jro6ml/spack-src to /tmp/culpo/spack-stage/spack-stage-lua-5.3.5-mru76cqdlrwasvbhutudefmwr7jro6ml/spack-src/luarocks/luarocks
==> [2021-03-22-10:22:20.456022] '/usr/bin/patch' '-s' '-p' '1' '-i' '/tmp/culpo/spack-stage/spack-stage-rj33fbjf/pkg-config.patch' '-d' '.'
==> [2021-03-22-10:22:20.463220] Applied patch http://lua.2524044.n2.nabble.com/attachment/7666421/0/pkg-config.patch
```
I just reverted the change to:
```diff
diff --git a/var/spack/repos/builtin/packages/lua/package.py b/var/spack/repos/builtin/packages/lua/package.py
index 600421c5cb..4b7ad53cbb 100644
--- a/var/spack/repos/builtin/packages/lua/package.py
+++ b/var/spack/repos/builtin/packages/lua/package.py
@@ -39,7 +39,12 @@ class Lua(Package):
     # luarocks needs unzip for some packages (e.g. lua-luaposix)
     depends_on('unzip', type='run')
 
-    patch('add-lua-pc-file.patch', when="+pcfile")
+    patch(
+        "http://lua.2524044.n2.nabble.com/attachment/7666421/0/pkg-config.patch",
+        sha256="208316c2564bdd5343fa522f3b230d84bd164058957059838df7df56876cb4ae",
+        when="+pcfile"
+    )
+
     resource(
         name="luarocks",
         url="https://keplerproject.github.io/luarocks/releases/"
```

Reply:
@v Another question: having this patch would interfere with anything depending on `lua`? If not we may add the patch without adding the variant, just as a fix to the `lua` package.

Reply:
I don't know the answer to that question, so to be conservative we probably shouldn't apply it globally. If someone needs that file for package config they can use the variant - so we could keep on the look out for that and assess how often it's needed (and then possibly make it default).

Reply:
Fair enough. If we determine that this can be always applied we can remove the variant later and reduce tha amount of things a user needs to think about when installing.